platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -906,41 +935,91 @@ int _odp_ipsec_sa_replay_precheck(ipsec_sa_t *ipsec_sa, uint32_t seq,
 int _odp_ipsec_sa_replay_update(ipsec_sa_t *ipsec_sa, uint32_t seq,
 				odp_ipsec_op_status_t *status)
 {
-	int cas = 0;
-	uint64_t state, new_state;
+	uint32_t bucket, wintop_bucket, new_bucket;
+	uint32_t bkt_diff, bkt_cnt;
+	uint64_t bit = 0;
 
 	if (!ipsec_sa->antireplay)
 		return 0;
 
-	state = odp_atomic_load_u64(&ipsec_sa->hot.in.antireplay);
+	if ((seq + ipsec_sa->hot.in.win_size) <=
+	    (odp_atomic_load_u64(&ipsec_sa->hot.in.wintop_seq) & 0xffffffff)) {
+		status->error.antireplay = 1;
+		return -1;
+	}
+
+	/* Window update for ws equal to 32 */
+	if (ipsec_sa->hot.in.win_size == IPSEC_AR_WIN_SIZE_MIN) {
```

Comment:
I think it would be good to separate these two totally different anti-replay codes in to their own functions to improve the clarity of the code.

Reply:
Ack, will separate out into two different functions, thinking of the below two

-  ipsec_ws32_replay_update()  -- for WS 32 case
-  ipsec_largews_replay_update()  -- for all other window sizes till 4096.